### PR TITLE
feat(picker-column): assign custom aria-labels to column options

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Developers cannot assign an `aria-label` to the individual button items in a picker column.

Issue URL: 

## What is the new behavior?

- Developers can optionally assign an `ariaLabel` to a picker column option, to assign the `aria-label` to the rendered button.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Note: Stencil has something strange happening with the spec tests. Since the picker column uses timeouts, you will receive an exception if the tests are located in the same file. The issue doesn't occur if you place the spec tests in separate files. I experimented with Jest timer mocks and this appears to be a Stencil issue. The team has asked I create a minimal reproduction for them to dig into, which will require additional time and isn't a blocker for this feature.

### CODE REVIEW

**Review by amandaejohnston — APPROVED** [review]

File: core/src/components/picker-column/test/picker-column-aria.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import { PickerColumnCmp } from '../picker-column';
+/**
+ * Stencil has an issue where having multiple spec tests in the same file,
+ * will cause an exception to be thrown. This test is located in a separate file
+ * to avoid this issue.
+ *
+ * Example exception:
+ * ```
+ * TypeError: Cannot read properties of undefined (reading '$instanceValues$')
+    at getValue (/Users/sean/Documents/ionic/framework-next/core/node_modules/@stencil/core/internal/testing/index.js:542:282)
```
Comment (liamdebeasi) [inline core/src/components/picker-column/test/picker-column-aria.spec.tsx:13]: Can we link to the Stencil issue instead of hardcoding the error?

**Review by liamdebeasi — APPROVED** [review]
One small change, but otherwise this looks great.

File: core/src/components/picker-column/test/picker-column-aria.spec.tsx (hunk 1 above)
Comment (sean-perkins) [reply core/src/components/picker-column/test/picker-column-aria.spec.tsx:13]: There currently isn't an issue logged to link to. I need to create a reproduction that does the same thing as `ion-picker-column` (using timeouts and executing code within the `connectedCallback`), so the team can diagnose and debug further.

Comment (liamdebeasi) [reply core/src/components/picker-column/test/picker-column-aria.spec.tsx]: Thank you!

**Pull request merged**